specification/ml/_types/Detector.ts

```diff
@@ -50,7 +54,7 @@ export class Detector {
   /**
    * The analysis function that is used. For example, `count`, `rare`, `mean`, `min`, `max`, or `sum`.
    */
-  function: string
+  function?: string
```

Comment:
>  function?: string

Is this actually optional?  I'm surprised if you can have a detector without a function. This should likely be reviewed by someone like @droberts195

Reply:
IMO it doesn't hurt to add this to the auto-generated docs and is probably harmless enough that it's not worth trying to introduce exceptional handling.